
%preamble
\documentclass[11pt, mathserif, aspectratio=169]{beamer}
\usetheme{Boadilla}
\usefonttheme{serif}

\addtobeamertemplate{frametitle}{\vspace*{0.2in}\hspace*{0.5in}}{\vspace*{0.3cm}}
\setbeamersize{text margin left=.7in,text margin right=.7in}
\setbeamercolor{author in head/foot}{fg=black!50!white, bg=white}
\setbeamercolor{title in head/foot}{fg=black!50!white, bg=white}
\setbeamerfont{footline}{size=\scriptsize,shape=\itshape}
\makeatother
\setbeamertemplate{footline}
{
  \leavevmode%
  \hbox{%
  \begin{beamercolorbox}[wd=.3\paperwidth,ht=2.5ex,dp=1.5ex,center]{author in head/foot}%
    \usebeamerfont{author in head/foot}\insertshortauthor
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.7\paperwidth,ht=2.5ex,dp=1.5ex,center]{title in head/foot}%
    \usebeamerfont{title in head/foot}\insertshorttitle\hspace*{3em}
    \insertframenumber{} / \inserttotalframenumber\hspace*{1.5ex}
  \end{beamercolorbox}}%
  \vskip0pt%
}
\makeatletter
\setbeamertemplate{navigation symbols}{}

%% Method to add QR code onto every slide
%\addtobeamertemplate{footline}{%
%  \setlength\unitlength{1ex}%
%  \begin{picture}(0,0) 
%    % \put{} defines the position of the frame
%    \put(132,55){\makebox(0,0)[br]{
%    \includegraphics[scale=.045]{pics/QG2019-qr-code.png}
%    }}%
%  \end{picture}%
%}{}

\usepackage{amsmath, amsthm, graphicx, amssymb}
\usepackage{color}
\usepackage{colortbl}
\usepackage{setspace}
\usepackage{tabularx,booktabs,adjustbox}
\usepackage{pifont}
\usepackage{tikz}
\usetikzlibrary{calc,fit,arrows,decorations.pathmorphing,decorations.footprints,backgrounds,fit,positioning}
\usetikzlibrary{shapes.symbols}
\usepackage[style=authoryear]{biblatex}
%\usepackage{cellspace}
%\setlength\cellspacetoplimit{1.5ex}
%\setlength\cellspacebottomlimit{1.5ex}

% Define a colour-blind-friendly palette.
%
% From:
% http://wiki.stdout.org/rcookbook/Graphs/Colors%20%28ggplot2%29/
\definecolor{Cblack}{rgb}{0,0,0}
\definecolor{Corange}{rgb}{0.9,0.6,0}
\definecolor{Cskyblue}{rgb}{0.35,0.7,0.9}
\definecolor{Cbluegreen}{rgb}{0,0.6,0.5}
\definecolor{Cyellow}{rgb}{0.95,0.9,0.25}
\definecolor{Cblue}{rgb}{0,0.45,0.7}
\definecolor{Cvermillion}{rgb}{0.8,0.4,0}
\definecolor{Cpurple}{rgb}{0.8,0.6,0.7}

% Define other colours.
\definecolor{grey}{gray}{0.8}

% Convenient colouring commands.
\newcommand{\red}[1]{\textcolor{red}{#1}}
\newcommand{\green}[1]{\textcolor{green}{#1}}
\newcommand{\blue}[1]{\textcolor{blue}{#1}}
\newcommand{\grey}[1]{\textcolor{grey}{#1}}
\newcommand{\white}[1]{\textcolor{white}{#1}}
\newcommand{\Cblack}[1]{\textcolor{Cblack}{#1}}
\newcommand{\Corange}[1]{\textcolor{Corange}{#1}}
\newcommand{\Cskyblue}[1]{\textcolor{Cskyblue}{#1}}
\newcommand{\Cbluegreen}[1]{\textcolor{Cbluegreen}{#1}}
\newcommand{\Cyellow}[1]{\textcolor{Cyellow}{#1}}
\newcommand{\Cblue}[1]{\textcolor{Cblue}{#1}}
\newcommand{\Cvermillion}[1]{\textcolor{Cvermillion}{#1}}
\newcommand{\Cpurple}[1]{\textcolor{Cpurple}{#1}}
\newcommand{\magenta}[1]{\textcolor{magenta}{#1}}
\newcommand{\gray}[1]{\textcolor{gray}{#1}}


%commands
\newcommand{\lt}{\left(}
\newcommand{\ls}{\left[}
\newcommand{\rt}{\right)}
\newcommand{\rs}{\right]}
\newcommand{\ts}{\thinspace}
\newcommand{\emp}[0]{\it}
\newcommand{\R}{\mathbb{R}}
\newcommand{\N}{\mathbb{N}}
\newcommand{\Var}{\mathrm{Var}}
\newcommand{\ds}{\displaystyle}
\newcommand{\be}{\beta}
\newcommand{\al}{\alpha}
\newcommand{\ga}{\gamma}
\newcommand{\p}{\mathbf{P}}
\newcommand{\E}{\mathbf{E}}
\newcommand{\Vari}{\mathrm{Var}}
\newcommand{\Cov}{\mathrm{Cov}}
\newcommand{\Corr}{\mathrm{Corr}}
\newcommand{\as}{\overset{a.s.}{\longrightarrow}}
\newcommand{\distn}{\overset{d}{\longrightarrow}}
% Checks and crosses
\newcommand{\crossmark}{\ding{55}}

% Rotating column headings in tables
\newcolumntype{R}[2]{%
    >{\adjustbox{angle=#1,lap=\width-(#2)}\bgroup}%
    l%
    <{\egroup}%
}
\newcommand*\rot{\multicolumn{1}{R{30}{1em}}}% no optional argument here, please!
% https://tex.stackexchange.com/questions/32683/rotated-column-titles-in-tabular

\AtBeginSection[]{
  \begin{frame}
  \vfill
  \centering
  \begin{beamercolorbox}[sep=8pt,center,shadow=true,rounded=true]{title}
    \usebeamerfont{title}\insertsectionhead\par%
  \end{beamercolorbox}
  \vfill
  \end{frame}
}

% environments
\newenvironment{wideitemize}{\itemize\addtolength{\itemsep}{10pt}}{\enditemize}


% packages
\usepackage{booktabs, hyperref}

% title
\titlegraphic{\includegraphics[scale=.5]{pics/unimelb-logo.jpg}}
\title{Efficient simulation of introgression,\\ admixture and local ancestry}
\author{Georgia Tsambos}
\institute{{\normalsize University of Melbourne, Australia}}
\date{Quantitative Genomics, 10 June 2019}
% information

% tikz colour settings
\tikzset{pop1/.style={blue!40},pop2/.style={red!40}}


% start of document

\begin{document}

\maketitle

\begin{frame}
\frametitle{Talk outline}

\begin{enumerate}
\itemsep7mm
\item[1. ] Introduction to admixture and local ancestry
\item[2. ] Tree sequences
\item[3. ] Simulating local ancestry with tree sequences: existing methods
\item[4. ] Simulating local ancestry with tree sequences: new method
\end{enumerate}
\end{frame}

\section{Intro to admixture and local ancestry}


\begin{frame}
\frametitle{What is admixture?}
%\begin{minipage}[t][2cm][t]{\textwidth}
%\begin{center}
%\input{pics/split-popgen.tex}
%\end{center}
%\end{minipage}

\begin{wideitemize}
%\item In the context of my work: {\it a group of people.}
%\item {\it Recombination, mutation, selection and drift} act over time to differentiate  populations that have split from each other.
%\item Analysing these differences are the basis for methods that infer the ancestry of DNA from people with mixed, or unknown backgrounds...
    \item An organism has \magenta{ancestry} with a given population if they have inherited some genetic material from ancestors who belonged to that population. 
    \item Any organism with $>1$ ancestry is \magenta{admixed}.
    \item \magenta{Introgression} is admixture between different species.
\end{wideitemize}

\end{frame}

%\begin{frame}
%\frametitle{What is admixture?}
%\begin{minipage}[t][2cm][t]{\textwidth}
%\begin{center}
%\input{pics/admix-popgen.tex}
%\end{center}
%\end{minipage}
%
%%{\small
%%\begin{wideitemize}
%%    \item A person has {\bf ancestry} with a given population if they have inherited some genetic material from ancestors who belonged to that population. 
%%    \item Any person with $>1$ ancestry is {\bf admixed}.
%%    \item Migrations that bring together historically distinct populations are {\bf admixture~events}.
%%\end{wideitemize}
%%}
%\end{frame}

\begin{frame}
%\frametitle{What is admixture?}
\begin{center}
\input{pics/admixed-family-tree.tex}
\end{center}
\end{frame}


\begin{frame}
\frametitle{Reporting ancestry}
\begin{center}
\input{pics/global-and-local-ancestry.tex}
\end{center}
My PhD work is about \magenta{simulating} and inferring local ancestry.\\[2mm]

%Understanding local ancestry is important in studies of demography and history, medicine and genetic pipelines.
\end{frame}

\begin{frame}
\frametitle{Being able to simulate genetic ancestry is useful!}
\begin{wideitemize}
\item {\bf Benchmarking and evaluating method performance\\}
To assess the accuracy of methods that infer ancestry, we need test datasets in which we know the true local ancestry.
\item {\bf Model training\\}
Some methods for ancestry inference are trained on simulated data.
\item {\bf Exploration\\}
Simulations allow us to explore the influence of various historical scenarios on observed patterns of genetic variation and inheritance.
\end{wideitemize}
\end{frame}

%\begin{frame}
%\frametitle{Why is understanding admixture and ancestry important?}
%
%\begin{wideitemize}
%\item {\bf Demography and history\\} Inference about the dates and composition of historical migrations and admixture events. 
%\item {\bf Medicine\\} GWAS and risk prediction studies, admixture mapping studies.
%\item {\bf Genetic pipelines\\} Phasing and imputation.
%\end{wideitemize}
%
%\end{frame}


\section{Tree sequences: the data format}

\begin{frame}
\frametitle{Context: genetic data is BIG and REPETITIVE}
 \begin{center}
{\tt
%\ldots GTAACGCGATAAGAGATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGAGATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGAGATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGAGATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGAAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGTAATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGTAATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGTAATAGCGTA\ldots \\
\ldots GTAACGCGATAAGATATTAGCCCAAAAACACAGACATGGTAATAGCGTA\ldots \\[3mm]
}
\gray{\footnotesize $\leftarrow 5\times 10^7$ bases for small human chromosome $\rightarrow$}
\end{center}
Repeated haplotypes are often just a consequence of shared history.\\[3mm]
\magenta{\bf Q. Can we use this history to represent DNA sequences more compactly?}
\end{frame}

%\begin{frame}
%\frametitle{Context - genetic data is BIG and REPETITIVE}
%\begin{center}
%{\tt
%%\ldots GTAACGCGATAAGA\Cvermillion{G}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%%\ldots GTAACGCGATAAGA\Cvermillion{G}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cvermillion{G}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cvermillion{G}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cvermillion{A}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cskyblue{T}AATAGCGTA\ldots \\
%%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cskyblue{T}AATAGCGTA\ldots \\
%%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cskyblue{T}AATAGCGTA\ldots \\
%\ldots GTAACGCGATAAGA\Cskyblue{T}ATTAGCCCAAAAACACAGACATGG\Cskyblue{T}AATAGCGTA\ldots \\
%}
%\end{center}
%Shared haplotypes are often due to shared history.\\[3mm]
%
%\magenta{\bf Q. Can we use this history to store DNA more compactly?}
%\end{frame}


%\begin{frame}
%\frametitle{Tree sequences are the future!}
%\begin{wideitemize}
%\item Tree sequences contain rich information about the \magenta{history} of a sample, not just the genotypes.
%\item We can \magenta{simulate} this data structure using well-established software  (\texttt{msprime}, \texttt{SLiM}). (With some caveats...)
%\item We can \magenta{infer} this data structure with some success (\texttt{tsinfer}).
%%\item In Kelleher et al. (2018), a tree-sequence holding an estimated history of chrom 22 for $\approx$ 500K individuals in the UK Biobank is constructed. It is of comparable size to the corresponding BCF!  
%\end{wideitemize}
%\end{frame}

\begin{frame}
\frametitle{Trees show genealogy at a single locus}
\begin{minipage}{.48\textwidth}
\begin{center}
\input{pics/simple-tree.tex}
\end{center}
\end{minipage}\hfill
\begin{minipage}{.48\textwidth}
\begin{wideitemize}
\item Nodes = haplotypes
\item Edge heights = time
\end{wideitemize}
\end{minipage}
\end{frame}

\begin{frame}
\frametitle{Tree sequences show genealogy over an interval of loci}
\begin{center}
\input{pics/tree-sequence-example/tree-sequence-example-1.tex}
\end{center}
%\begin{wideitemize}
%\item 
%The set of genealogies over the entire chromosome can be recorded in a \magenta{tree sequence}.
%\end{wideitemize}
\end{frame}

\begin{frame}
\frametitle{Tree sequences can encode haplotypes}
\begin{center}
\input{pics/tree-sequence-example/tree-sequence-example-2.tex}
\end{center}
\end{frame}

\begin{frame}
\frametitle{Tree sequences can hold info on local ancestry}
\begin{minipage}{.38\textwidth}
\begin{center}
\input{pics/tree-with-populations.tex}
\end{center}
\end{minipage}\hfill
\begin{minipage}{.58\textwidth}
Nodes may be assigned to populations.\\
The branch joining a sample node to an ancestral node shows the sample's ancestry.
\end{minipage}
\end{frame}

%\begin{frame}
%\frametitle{Tree sequences are good, use them}
%\begin{wideitemize}
%\item $>100\times$ smaller than VCF files for simulated datasets!
%\item Quick to load and query
%\item \texttt{tskit}: a collection of software for creating and analysing tree sequence data
%\end{wideitemize}
%\end{frame}

%\begin{frame}
%\frametitle{Tree sequences can be stored in tables}
%\hspace{-1.5cm}
%\begin{minipage}{.48\textwidth}
%\begin{center}
%\input{pics/tree-sequence-example/tree-sequence-example-3.tex}
%\end{center}
%\end{minipage}\hfill
%\begin{minipage}{.48\textwidth}
%\begin{center}
%\input{pics/tree-sequence-example/tree-sequence-example-3-table.tex}
%\end{center}
%\end{minipage}
%\end{frame}

%\begin{frame}
%\frametitle{Local ancestry can be extracted from tree sequences}
%\begin{center}
%\input{pics/local-ancestry-extraction.tex}
%\end{center}
%\end{frame}

%\begin{frame}
%\frametitle{But doing this efficiently is hard...}
%\begin{wideitemize}
%\item There are substiantial correlations in genealogy between individual nodes, and across genomes.
%\item Requires clever operations for altering tree sequences by `simplifying' the tables that represent them.
%\item More detail in an upcoming paper by Georgia, Damjan, Stephen, Jerome Kelleher (Oxford) \& Peter Ralph (Oregon).
%\end{wideitemize}
%\end{frame}

\section{Simulating admixture: existing methods}

\begin{frame}
\frametitle{\texttt{msprime} simulates \magenta{a sample} backwards-in-time}
\begin{minipage}{.48\textwidth}
\input{pics/msprime/msprime-0.tex}
\end{minipage}\hfill
\begin{minipage}{.48\textwidth}
%\begin{wideitemize}
Simulates tree sequences under the coalescent model.\\[5mm]
%\end{wideitemize}
%\gray{\scriptsize Kelleher, J., Etheridge, A. M., \& McVean, G. (2016). Efficient Coalescent Simulation and Genealogical Analysis for Large Sample Sizes. PLOS Computational Biology, 12(5).}
\end{minipage}
\end{frame}

\begin{frame}
\frametitle{\texttt{msprime} simulates \magenta{a sample} backwards-in-time}
\begin{minipage}{.48\textwidth}
\input{pics/msprime/msprime-1.tex}
\end{minipage}\hfill
\begin{minipage}{.48\textwidth}
%\begin{wideitemize}
Simulates tree sequences under the coalescent model.\\[5mm]
%\end{wideitemize}
%\gray{\scriptsize Kelleher, J., Etheridge, A. M., \& McVean, G. (2016). Efficient Coalescent Simulation and Genealogical Analysis for Large Sample Sizes. PLOS Computational Biology, 12(5).}
\end{minipage}
\end{frame}

\begin{frame}
\frametitle{\texttt{msprime} simulates \magenta{a sample} backwards-in-time}
\begin{minipage}{.48\textwidth}
\input{pics/msprime/msprime-2.tex}
\end{minipage}\hfill
\begin{minipage}{.48\textwidth}
%\begin{wideitemize}
Simulates tree sequences under the coalescent model.\\[5mm]
%\end{wideitemize}
%\gray{\scriptsize Kelleher, J., Etheridge, A. M., \& McVean, G. (2016). Efficient Coalescent Simulation and Genealogical Analysis for Large Sample Sizes. PLOS Computational Biology, 12(5).}
\end{minipage}
\end{frame}

\begin{frame}
\frametitle{\texttt{msprime} simulates \magenta{a sample} backwards-in-time}
\begin{minipage}{.48\textwidth}
\input{pics/msprime/msprime-3.tex}
\end{minipage}\hfill
\begin{minipage}{.48\textwidth}
%\begin{wideitemize}
Simulates tree sequences under the coalescent model.\\[5mm]
%\end{wideitemize}
%\gray{\scriptsize Kelleher, J., Etheridge, A. M., \& McVean, G. (2016). Efficient Coalescent Simulation and Genealogical Analysis for Large Sample Sizes. PLOS Computational Biology, 12(5).}
\end{minipage}
\end{frame}

\begin{frame}
\frametitle{A simplified toy example: Neanderthal introgression}
\begin{center}
\begin{tabularx}{1\textwidth}{p{3cm}X}
\toprule
{\bf Generations} & {\bf Event}\\
\midrule
$\approx$ 240\ts 000 & Common ancestor of all modern Eurasians and Neanderthals at all loci.\\[1mm]
20\ts 000 & Divergence of Eurasians and Neanderthals.\\[1mm]
2\ts 500 & 2\% introgression of Neanderthals into Eurasians.\\[1mm]
0 & Samples from 100 Eurasian individuals obtained.\\
\bottomrule
\end{tabularx}
\end{center}
Chromosome of 50\ts 000\ts 000 base pairs, constant effective population sizes of 5000 individuals, uniform recombination rate $1\times 10^{-8}$ bp/generation, uniform mutation rate $1\times 10^{-8}$ bp/generation, all variation is neutral.
\end{frame}

\begin{frame}
\frametitle{\texttt{msprime} results: missing ancestry data}
\includegraphics[scale=.22]{pics/msprime/msprime-sample.png}\quad\quad
Global ancestry averaged over all of the simulated samples was 
\begin{itemize}
\item $96.0\%$ Sapiens
\item $<0.05\%$ Neanderthal
\item \magenta{$4.0\%$ unassigned.}
\end{itemize}
%0.960 Sapiens, 0.000 Neanderthal, \magenta{0.040 unassigned.}
\end{frame}

\begin{frame}
\frametitle{Missing ancestry is due to incomplete lineage sorting}
\begin{center}
\input{pics/simulation-problem.tex}
\end{center}
Some samples do not have simulated ancestors in the given populations.
\end{frame}

\begin{frame}
\frametitle{\texttt{msprime} performance}
\begin{center}
\small
\centering
\setlength{\aboverulesep}{5pt}
\setlength{\belowrulesep}{5pt}
\begin{tabularx}{1\textwidth}{p{3.5cm}p{2cm}p{1.9cm}p{1.9cm}X}
\toprule
%& \multicolumn{3}{c}{{\bf Global ancestry}} \\[1mm]
 & Missing data & Run time & File size & Realism \\
\midrule 
{\bf default} & $4.0$\% & 6 sec &  9 Mb & \crossmark \\ \addlinespace
{\bf + all ancestors} & $0.0$\% & 53 sec  &  1700 Mb & \crossmark \\ \addlinespace
%{\bf + migration records} & $0.0$\% &  & & \crossmark \\ \addlinespace
\bottomrule
\end{tabularx}
\end{center}
\vspace{5mm}
Restricted by the limitations of the coalescent model: no selection, random mating, small sample sizes, no more than 1 mutation at any location \ldots
\end{frame}

%\begin{frame}
%\frametitle{}
%\begin{wideitemize}
%\item \texttt{msprime} simulates the sparsest set of haplotypes that are needed to represent the relative genealogical history of the samples.
%\item This sparsity makes it quick and memory-efficient to run.
%\item Limited by the assumptions of the coalescent model: infinite sites, no selection or deviations from random mating, small sample size relative to effective population size...
%\item See (Kelleher et al., 2016) for more details.
%\end{wideitemize}
%\end{frame}

\begin{frame}
\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
\begin{minipage}{.58\textwidth}
\input{pics/slim/slim-0.tex}
\end{minipage}\hfill
\begin{minipage}{.4\textwidth}
Alternates between
\begin{enumerate}
\item forward simulation
\item pruning of irrelevant history.
\end{enumerate}
\end{minipage}
\end{frame}

%\begin{frame}
%\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
%\begin{minipage}{.58\textwidth}
%\input{pics/slim/slim-1.tex}
%\end{minipage}\hfill
%\begin{minipage}{.4\textwidth}
%Alternates between
%\begin{enumerate}
%\item \magenta{forward simulation}
%\item pruning of irrelevant history.
%\end{enumerate}
%\end{minipage}
%\end{frame}

\begin{frame}
\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
\begin{minipage}{.58\textwidth}
\input{pics/slim/slim-2.tex}
\end{minipage}\hfill
\begin{minipage}{.4\textwidth}
Alternates between
\begin{enumerate}
\item forward simulation
\item pruning of irrelevant history.
\end{enumerate}
\end{minipage}
\end{frame}

%\begin{frame}
%\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
%\begin{minipage}{.58\textwidth}
%\input{pics/slim/slim-3.tex}
%\end{minipage}\hfill
%\begin{minipage}{.4\textwidth}
%Alternates between
%\begin{enumerate}
%\item \magenta{forward simulation}
%\item pruning of irrelevant history.
%\end{enumerate}
%\end{minipage}
%\end{frame}

\begin{frame}
\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
\begin{minipage}{.58\textwidth}
\input{pics/slim/slim-4.tex}
\end{minipage}\hfill
\begin{minipage}{.4\textwidth}
Alternates between
\begin{enumerate}
\item forward simulation
\item pruning of irrelevant history.
\end{enumerate}
\end{minipage}
\end{frame}

%\begin{frame}
%\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
%\begin{minipage}{.58\textwidth}
%\input{pics/slim/slim-5.tex}
%\end{minipage}\hfill
%\begin{minipage}{.4\textwidth}
%Alternates between
%\begin{enumerate}
%\item \magenta{forward simulation}
%\item pruning of irrelevant history.
%\end{enumerate}
%\end{minipage}
%\end{frame}

\begin{frame}
\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
\begin{minipage}{.58\textwidth}
\input{pics/slim/slim-6.tex}
\end{minipage}\hfill
\begin{minipage}{.4\textwidth}
Alternates between
\begin{enumerate}
\item forward simulation
\item pruning of irrelevant history.
\end{enumerate}
\end{minipage}
\end{frame}

%\begin{frame}
%\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
%\begin{minipage}{.58\textwidth}
%\input{pics/slim/slim-7.tex}
%\end{minipage}\hfill
%\begin{minipage}{.4\textwidth}
%Alternates between
%\begin{enumerate}
%\item \magenta{forward simulation}
%\item pruning of irrelevant history.
%\end{enumerate}
%\end{minipage}
%\end{frame}

\begin{frame}
\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
\begin{minipage}{.58\textwidth}
\input{pics/slim/slim-8.tex}
\end{minipage}\hfill
\begin{minipage}{.4\textwidth}
Alternates between
\begin{enumerate}
\item forward simulation
\item pruning of irrelevant history.
\end{enumerate}
\end{minipage}
\end{frame}

%\begin{frame}
%\frametitle{\texttt{SLiM} simulates \magenta{an entire population} forward-in-time}
%\begin{minipage}{.58\textwidth}
%\input{pics/slim/slim-10.tex}
%\end{minipage}\hfill
%\begin{minipage}{.4\textwidth}
%Mutations can be added during the simulation of the tree topologies, or generated independently and added afterwards.
%\end{minipage}
%\end{frame}

\begin{frame}
\frametitle{\texttt{SLiM} performance}
\begin{center}
\small
\centering
\setlength{\aboverulesep}{5pt}
\setlength{\belowrulesep}{5pt}
\begin{tabularx}{.9\textwidth}{p{2cm}p{2cm}p{1.9cm}p{1.9cm}X}
\toprule
%& \multicolumn{3}{c}{{\bf Global ancestry}} \\[1mm]
 & Missing data & Run time & File size & Realism \\
\midrule 
%{\bf default} & $4.0$\% & 6 sec &  9 Mb & \crossmark \\ \addlinespace
\texttt{msprime}& $0.0$\% & 53 sec  &  1700 Mb & \crossmark \\ \addlinespace
\magenta{\texttt{SLiM}} & \magenta{$0.0$\%} & \magenta{$>1$ hr} & \magenta{41 Mb} & \magenta{\checkmark}  \\
\bottomrule
\end{tabularx}
\end{center}
\vspace{5mm}
\end{frame}

\section{Simulating tree sequences: new method}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-0.tex}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-1.tex}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-2.tex}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-3.tex}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-5.tex}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-4.tex}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\centering
\input{pics/simulation-method/simulation-method-6.tex}
\end{center}
\end{frame}

\begin{frame}
\frametitle{New method performance}
\begin{center}
\small
\centering
\setlength{\aboverulesep}{5pt}
\setlength{\belowrulesep}{5pt}
\begin{tabularx}{1\textwidth}{p{3.5cm}p{2cm}p{1.9cm}p{1.9cm}X}
\toprule
%& \multicolumn{3}{c}{{\bf Global ancestry}} \\[1mm]
 & Missing data & Run time &  File size & Realism \\
\midrule 
%{\bf \texttt{msprime}} & $4.0$\% & 6 sec & 9 Mb & \crossmark \\ \addlinespace
{\bf \texttt{msprime}}  & $0.0$\% & 53 sec & 1700 Mb  & \crossmark \\ \addlinespace
%{\bf + migration records} & $0.0$\% &  & & \crossmark \\ 
%\midrule
%{\bf \texttt{SLiM}} & $4.0$\% & 10 min & 16 Mb  & \checkmark \\ \addlinespace
{\bf \texttt{SLiM}}& {$0.0$\%} & $>1$ hr  &  41 Mb & {\checkmark} \\ \addlinespace
%\midrule
{\bf\magenta{Georgia}} & \magenta{$0.0$\%} & \magenta{86 sec} & \magenta{39 Mb}  & \magenta{\checkmark} \\ \addlinespace
\bottomrule
\end{tabularx}
\end{center}
\vspace{5mm}
\end{frame}


%\section{Q. Can these methods simulate local ancestry?}
%
%
%
%
%\begin{frame}
%\frametitle{\texttt{SLiM} results}
%\includegraphics[scale=.22]{pics/slim-sample.png}\quad\quad
%Global ancestry averaged over all of the simulated samples was 
%\begin{itemize}
%\item $96.2\%$ Sapiens
%\item $<0.05\%$ Neanderthal
%\item \magenta{$3.8\%$ unassigned.}
%\end{itemize}
%%0.960 Sapiens, 0.000 Neanderthal, \magenta{0.040 unassigned.}
%\end{frame}
%
%\begin{frame}
%\frametitle{\texttt{SLiM} results - missing ancestry}
%\includegraphics[scale=.22]{pics/slim-sample-missing.png}\quad\quad
%Global ancestry averaged over all of the simulated samples was 
%\begin{itemize}
%\item $96.2\%$ Sapiens
%\item $<0.05\%$ Neanderthal
%\item \magenta{$3.8\%$ unassigned.}
%\end{itemize}
%%0.960 Sapiens, 0.000 Neanderthal, \magenta{0.040 unassigned.}
%\end{frame}
%
%
%
%%\section{A new method for simulating admixture}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-0.tex}
%%\end{center}
%%\end{frame}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-1.tex}
%%\end{center}
%%\end{frame}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-2.tex}
%%\end{center}
%%\end{frame}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-3.tex}
%%\end{center}
%%\end{frame}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-5.tex}
%%\end{center}
%%\end{frame}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-4.tex}
%%\end{center}
%%\end{frame}
%%
%%\begin{frame}
%%\begin{center}
%%\centering
%%\input{pics/simulation-method/simulation-method-6.tex}
%%\end{center}
%%\end{frame}
%
%\begin{frame}
%\frametitle{Results}
%\includegraphics[scale=.22]{pics/new-method-sample.png}\quad\quad
%Global ancestry averaged over all of the simulated samples was 
%\begin{itemize}
%\item $98.0\%$ Sapiens
%\item $2.0\%$ Neanderthal
%\item \magenta{$0.0\%$ unassigned.}
%\end{itemize}
%%0.960 Sapiens, 0.000 Neanderthal, \magenta{0.040 unassigned.}
%\end{frame}
%
%\section{Why this work matters}
%
%\begin{frame}
%\begin{center}
%\small
%\centering
%\begin{tabularx}{.7\textwidth}{p{2.5cm}p{1.7cm}p{1.7cm}X}
%\toprule
%& \multicolumn{3}{c}{{\bf Global ancestry}} \\[1mm]
% & { Sapiens} & { Neanderthal} &  Missing \\
%\midrule 
%{\bf msprime} & $96.0$\% & $\approx 0.0$\% & $4.0\%$  \\[3mm]
%{\bf SLiM} & $96.2\%$ &  $\approx 0.0\%$ & $3.8\%$ \\[3mm]
%{\bf Georgia} & $98.0\%$ & $2.0\%$ & $0.0\%$  \\[3mm]
%\midrule
%{\bf\magenta{Expected}} & \magenta{$98.0\%$} & \magenta{$2.0\%$} & \magenta{$0.0\%$} \\[1mm]
%\bottomrule
%\end{tabularx}
%\end{center}
%\vspace{5mm}
%Affects the computation of ancestral tract lengths, population-specific frequency spectra, ancestry-informative markers...
%\end{frame}

%\begin{frame}
%\frametitle{Simulations are often used to benchmark methods\\ for inferring local ancestry}
%{\footnotesize
%Steinruecken, M., Spence, J. P., Kamm, J. A., Wieczorek, E., \& Song, Y. S. (2018). Model-based detection and analysis of introgressed Neanderthal ancestry in modern humans. \emph{Molecular Ecology, 27(19)}, pp. 3873 - 3888.
%}\\[2mm]
%
%\begin{center}
%\includegraphics[scale=.5]{pics/paper-2018Steinrucken/2018Steinrucken-1.png}\\[2mm]
%\includegraphics[scale=.5]{pics/paper-2018Steinrucken/2018Steinrucken-2.png}
%\end{center}
%\end{frame}
%
%\begin{frame}
%\frametitle{Testing your method on incomplete ancestral data\\
%makes it look better than it actually is}
%\begin{center}
%\includegraphics[scale=.5]{pics/paper-2018Steinrucken/2018Steinrucken-3.png}\\[10mm]
%\magenta{NOPE :(}
%\end{center}
%\end{frame}

%\begin{frame}
%\frametitle{Another example}
%{\small
%Browning, S. R., Browning, B. L., Zhou, Y., Tucci, S., \& Akey, J. M. (2018). Analysis of Human Sequence Data Reveals Two Pulses of Archaic Denisovan Admixture. \emph{Cell, 173(1),} 53–61
%}\\[2mm]
%\begin{center}
%\includegraphics[scale=.4]{pics/paper-2018Browning/2018Browning-1.png}\\[2mm]
%\includegraphics[scale=.4]{pics/paper-2018Browning/2018Browning-2.png}
%\end{center}
%This paper inferred a `true' introgression status of haplotypes simulated with msprime (?)
%
%\end{frame}

%\begin{frame}
%\frametitle{Ancestry inference methods trained on flawed simulations\\
%are likely to be biased
%}
%{\footnotesize
%Sankararaman, S. et al (2014). The genomic landscape of Neanderthal ancestry in present-day humans. \emph{Nature, 507(7492)}, pp. 354 - 357.
%}\\[2mm]
%\begin{center}
%\includegraphics[scale=.35]{pics/paper-2014Sankararaman/Sankararaman-1.png}\\[2mm]
%\includegraphics[scale=.4]{pics/paper-2014Sankararaman/Sankararaman-2.png}
%\end{center}
%\end{frame}

\begin{frame}
\frametitle{Summary}
\begin{wideitemize}
\item It is useful to keep track of local ancestry in genetic simulations.
\item Standard tree sequence simulators are able to do this with the help of efficient ancestry-extraction algorithms.
\item A new simulation method can do this quickly while also allowing the user to model complex admixture scenarios.
\end{wideitemize}
\end{frame}

\begin{frame}
\frametitle{Thanks to...}

\begin{tabularx}{.68\textwidth}{p{3cm}X}
My supervisors & $\begin{cases}
\mathrm{Damjan\ts\ts Vukcevic\ts\ts (University\ts\ts of\ts\ts Melbourne)} \\
\mathrm{Stephen\ts\ts Leslie\ts\ts (University\ts\ts of \ts\ts Melbourne)}
\end{cases}$\\ \addlinespace
My collaborators &
$\begin{cases}
\mathrm{Peter\ts\ts Ralph\ts\ts(University\ts\ts of\ts\ts Oregon)} \\
\mathrm{Jerome\ts\ts Kelleher\ts\ts(BDI,\ts\ts University\ts\ts of\ts\ts Oxford)}
\end{cases}$\\ \addlinespace
Sources of \$\ &
$\begin{cases}
\mathrm{Helen\ts\ts Freeman\ts\ts scholarship,\ts\ts UniMelb} \\
\mathrm{Maurice\ts\ts Belz\ts\ts Fund,\ts\ts UniMelb}\\
\mathrm{Research\ts\ts Training\ts\ts Scheme,\ts\ts Australian\ts\ts Government}
\end{cases}$\\ \addlinespace
\end{tabularx}

Special thanks to Jerome and the University of Oxford for hosting me in the UK this year.

%My supervisors\quad
\end{frame}

\begin{frame}
\frametitle{References}

{\bf \texttt{msprime} and tree sequences:}\\
Kelleher, J., et al. (2016). Efficient Coalescent Simulation and Genealogical Analysis for Large Sample Sizes. PLOS Computational Biology, 12(5).
\vspace{5mm}

{\bf \texttt{SLiM}:}\\
Galloway, J.,  et al. (2018). Tree-sequence recording in SLiM opens new horizons for forward-time simulation of whole genomes. Molecular Ecology Resources, (November 2018), 552–566.

\end{frame}

\end{document}